modules/lib/core/index.d.ts

```diff
@@ -52,14 +52,54 @@ export interface PublishInfo {
     first?: string;
 }
 
-export interface Component<Config extends object = object, Regions extends Record<string, Region> = Record<string, Region>> {
-    config: Config;
-    descriptor: string;
+export interface FragmentComponent {
+    fragment: string;
     path: string;
-    type: 'page' | 'layout' | 'part';
+    type: 'fragment'
+}
+
+export interface LayoutComponent<
+    Config extends object = object,
+    Regions extends Record<string, Region> = Record<string, Region>
+> {
+    config: Config
+    descriptor: string
+    path?: string // Missing in fragmentPreview https://github.com/enonic/xp/issues/10116
+    regions: Regions;
+    type: 'layout'
+}
+
+export interface PartComponent<Config extends object = object> {
+    config: Config
+    descriptor: string
+    path?: string // Missing in fragmentPreview https://github.com/enonic/xp/issues/10116
+    type: 'part'
+}
+
+export interface PageComponent<
+    Config extends object = object,
+    Regions extends Record<string, Region> = Record<string, Region>
+> {
+    config: Config
+    descriptor: string
+    path: '/'
     regions: Regions;
+    type: 'page'
 }
 
+export interface TextComponent {
+    path: string
+    text: string
+    type: 'text'
+}
+
+export type Component<Config extends object = object, Regions extends Record<string, Region> = Record<string, Region>> =
+    | FragmentComponent
+    | LayoutComponent<Config, Regions>
+    | PartComponent<Config>
+    | PageComponent
+    | TextComponent;
+
 export interface Region<Config extends object = object> {
```

Comment:
If definetly like the idea of passing Component rather than Config into Region, after all Fragment and Text components doesn't even have Config... It's a "breaking" change though.

Reply:
I think it's always better to pass in the full shapes of properties on the "root" level of an interface, or one will run into problems later on, like now, when it's a breaking change.

Reply:
For example

```
export interface Region<Components extends Component[] = object[]> {
    name: string;
    components: Components;
}
```

Or just

```
export interface Region<Components extends object[] = object[]> {
    name: string;
    components: Components;
}
```

Then maybe one could pass in even a tuple type. Say the fist component should always be text and the second a part. But that not really possible to restrict in Content Studio (yet?)

Reply:
Yea, it's not good that it's a breaking change. But this will probably not be an issue that happens more then a handful of times in a project.

Reply:
...unless you change the signaure of  `getComponent` to:

```typescript
export declare function getComponent<
  T extends PartComponent | LayoutComponent = PartComponent | LayoutComponent
>(): T | null;
```

Which would create breakage in many parts, but would be a very nice Interface to program against.

Reply:
Sidenote: I believe getComponent used to work in page controller aswell (pre xp7) https://github.com/enonic/xp/issues/7667

Reply:
I would change getComponent too :)

Reply:
Which also requires GetCurrentComponentHandler to be changed
```
interface GetCurrentComponentHandler<
    T extends PartComponent | LayoutComponent = PartComponent | LayoutComponent
> {
    execute(): T | null;
}
```

Reply:
Conversation continues in other conversation